### PR TITLE
Move get_new_folder to padertorch.io

### PR DESCRIPTION
**Pull request by boeddeker**
- Move `get_new_folder` to `padertorch.io`
    - add `id_type` argument to switch between `index` and `time`. Does someone have better names?
 - Add `get_basedir`
    - Does someone know a better name? This name is currently inspired from sacred.
    - Some information for the folder name convention:
      Let's say, we want that all information from one experiment is saved in the folder `.../storage/my_fance_experiment/2`.
       - The complete path is then the `storage_dir='.../storage/my_fance_experiment/2'`. This is saved as property of the trainer (`padertorch.Trainer.storage_dir`).
       - The `2` is interpreted as an experiment id. The first experiment get `1`, the second `2`, ...
       - The name `my_fance_experiment` is a name for your experiment. 
       - The enviroment variable `STORAGE_ROOT` is `'.../storage'`

### CODE REVIEW

**Comment by boeddeker** [comment]
add Lukas style of get_new_folder with adjective_color_animal

**Comment by jensheit** [comment]
Perhaps we should also rename "get_new_folder" since it is more something like "get_next_experiment_dir".

Instead of get_base_dir wouldn't it be get_experiment_dir() (especially true if we rename get_new_folder)?

**Comment by boeddeker** [comment]
I think `get_new_folder` is ok. But depending on the name for `get_base_dir` a renaming may be resonable.

I don't like the combination of `get_experiment_dir` and `get_next_experiment_dir`. The first one sound like, get an existing dir, while the second get a new dir. But the first we want to use to get the parent dir.

**Comment by boeddeker** [comment]
We had an offline discussion. Here our conclusion:
 - Rename `get_new_folder` to `get_new_subdir`
 - Move `get_new_subdir` to `paderbox`
 - Add cli option to `get_new_subdir`
 - Recommend a wrapper in `padertorch` instead of two functions.
   - As name we favored `get_new_storage_dir`
 - Remove `force_suffix` and `try_id` -> confusing
   - Instead introduce `prefix` and `suffix`.
 - Rename `id_type` to `id_naming`

**Comment by boeddeker** [comment]
Tests fail, because https://github.com/fgnt/paderbox/pull/18 has to be merged first.

File: padertorch/io.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            return simu_dir
+        except FileExistsError:
+            # Catch race conditions
+            if i > 100:
+                # After some tries,
+                # expect that something other went wrong
+                raise
+
+
+def get_basedir(
```
Comment (jensheit) [inline padertorch/io.py]: I do not think that we need this function here

**Comment by boeddeker** [comment]
https://github.com/fgnt/paderbox/pull/18 is now merged and this PR works with paderbox master.

File: padertorch/io.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+def get_new_storage_dir(
+        experiment_name,
+        *,
+        id_naming='index',
+        mkdir=True,
+        prefix=None,
+        suffix=None,
+        consider_mpi=False,
+        dry_run=False,
+):
```
Comment (thequilo) [inline padertorch/io.py:16]: Ads type annotations similar to the ones in paderbox?

File: padertorch/io.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    This is a wrapper around `paderbox.io.new_subdir.get_new_subdir`.
+
```
Comment (thequilo) [inline padertorch/io.py:19]: Copy the parameter docs from paderbox? I'm not sure if referencing it is enough. Especially the mpi warning is important

File: padertorch/io.py (hunk 2 above)
Comment (boeddeker) [reply padertorch/io.py:16]: Done

File: padertorch/io.py (hunk 3 above)
Comment (boeddeker) [reply padertorch/io.py]: Done

**Pull request merged**